packages/vite/src/node/packages.ts

```diff
@@ -194,10 +196,30 @@ export function loadPackageData(pkgPath: string): PackageData {
     hasSideEffects = () => null
   }
 
+  // Determine if the package is locally within a monorepo workspace
+  const inWorkspace = isPackageInWorkspace(pkgPath)
+  let srcDir = null
+
+  // When in a workspace, attempt to find a source directory.
+  // We do this check here, so that it only runs once per package,
+  // and not for every file in the package!
+  if (inWorkspace) {
+    for (const srcName of ['src', 'sources']) {
```

Comment:
I don't think Vite should be hardcoding these directories. The users should be free to arrange the output however they want.

Reply:
I agree. I originally started adding this behind a config setting, similar to file extensions, but pushed it off to get the POC working.